src/mono/wasm/runtime/method-binding.ts

```diff
@@ -559,7 +554,7 @@ export function mono_bind_method(method: MonoMethod, this_arg: null, args_marsha
         displayName += "_this" + this_arg;
 
     body.push(
-        `_teardown_after_call (${converterKey}, token, buffer, resultRoot, exceptionRoot, argsRootBuffer);`,
+        `_teardown_after_call (${converterKey}, token, buffer, resultRoot, exceptionRoot, sp);`,
```

Comment:
Possibly something could throw JS exception during marshaling and `_teardown_after_call` would not be called. That should be in `finally` so that return back to managed land would work even in that case.

Reply:
I originally designed this so bound methods don't use 'finally' since it reduces the performance of the compiled JS. I'm open to changing that in a follow-up PR if you're concerned about it